pygmt/tests/test_x2sys_cross.py

```diff
@@ -142,8 +143,9 @@ def test_x2sys_cross_input_dataframe_with_nan(tracks):
             tag=tag, fmtfile="xyz", suffix="xyzt", units=["de", "se"], force=True
         )
 
-        tracks[0].loc[tracks[0]["z"] < -15, "z"] = np.nan  # set some values to NaN
-        output = x2sys_cross(tracks=tracks, tag=tag, coe="i")
+        newtracks = copy.deepcopy(x=tracks)
+        newtracks[0].loc[newtracks[0]["z"] < -15, "z"] = np.nan  # set some NaN values
+        output = x2sys_cross(tracks=newtracks, tag=tag, coe="i")
```

Comment:
@seisman, this is similar to your commit f97d92efbafde01d951285577a6251dd5ac34269 in #2936, but I used a deep copy (`copy.deepcopy()`) instead of a shallow copy (`.copy()`) because setting NaNs on the shallow copy `newtracks` still modified the original `tracks` object.